fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/keyspace/LocatableResolver.java

```diff
@@ -423,26 +430,65 @@ public FDBDatabase getDatabase() {
     @VisibleForTesting
     public abstract CompletableFuture<Void> setWindow(long count);
 
-    protected abstract Subspace getStateSubspace();
+    protected abstract CompletableFuture<Subspace> getStateSubspace(FDBRecordContext context);
 
     /**
      * The {@link Subspace} where this resolver stores the mappings from <code>key</code> {@link String}s to
      * <code>value</code> {@link Long}. Direct access to this subspace is not needed by general users and extreme care
      * should be taken when interacting with it.
+     * @deprecated This creates a new context and blocks waiting for the result, instead use
+     * {@link #getMappingSubspace(FDBRecordContext)}.
      * @return The mapping subspace.
      */
     @Nonnull
-    public abstract Subspace getMappingSubspace();
+    @API(API.Status.DEPRECATED)
+    @Deprecated
+    public Subspace getMappingSubspace() {
+        try (FDBRecordContext context = database.openContext()) {
+            LOGGER.warn("blocking call to getMappingSubspace");
+            return context.join(getMappingSubspace(context));
+        }
+    }
+
+    /**
+     * Get a {@link CompletableFuture} that will contain the {@link Subspace} where this resolver stores
+     * the mappings from <code>key</code> {@link String}s to <code>value</code> {@link Long}. Direct access
+     * to this subspace is not needed by general users and extreme care should be taken when interacting with it.
+     * @param context The {@link FDBRecordContext} to use when looking up the mapping {@link Subspace} in the database.
+     * @return A future that, when ready, will hold the mapping subspace.
+     */
+    @Nonnull
+    public abstract CompletableFuture<Subspace> getMappingSubspace(FDBRecordContext context);
```

Comment:
Sorry I missed this on the last review.  I think we are settling on a convention that calls producing futures, should end in `Async`.  The advantage of this is that the changes I put in https://github.com/FoundationDB/fdb-record-layer/pull/324 will attempt to detect if any calls that are producing futures are, in the process, calling blocking API calls.